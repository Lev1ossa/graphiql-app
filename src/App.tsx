import { useEffect } from 'react'
import { RouterProvider } from 'react-router-dom'
import { auth } from './firebase'
import { router } from './router/router'
import { useBoundStore } from './store'

export function App(): React.ReactElement {
  const setUser = useBoundStore((state) => state.setUser)

  useEffect(() => {
    const unsubsrcibe = auth.onAuthStateChanged((user) => {
      setUser(user)
    })
    return unsubsrcibe
  }, [setUser])

  return <RouterProvider router={router} />
}
