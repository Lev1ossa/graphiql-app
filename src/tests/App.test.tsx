import { App } from '@/App'
import { render } from '@testing-library/react'
import { expect, test } from 'vitest'

test('Correct App snapshot', () => {
  const { baseElement } = render(<App />)
  expect(baseElement).toMatchSnapshot()
})
