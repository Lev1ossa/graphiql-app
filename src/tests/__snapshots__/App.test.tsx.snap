// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Correct App snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiStack-root css-1np9wzt-MuiStack-root"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-1k8pp9v-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="/welcome"
          >
            <div
              class="MuiBox-root css-10gearf"
            >
              <svg
                fill="none"
                height="90%"
                viewBox="0 0 38 43"
                width="90%"
              >
                <path
                  d="m1.873 31.326 1.552.896 17.29-29.946-1.552-.896-17.29 29.946Z"
                  fill="currentColor"
                />
                <path
                  d="M36.098 28.933H1.518v1.793h34.58v-1.793Z"
                  fill="currentColor"
                />
                <path
                  d="M2.202 29.94 19.5 39.927l.896-1.552-17.297-9.987-.896 1.552ZM17.228 3.913 34.524 13.9l.896-1.552L18.124 2.36l-.896 1.552Z"
                  fill="currentColor"
                />
                <path
                  d="m2.207 12.34.896 1.553L20.4 3.906l-.897-1.552-17.296 9.987Z"
                  fill="currentColor"
                />
                <path
                  d="m16.915 2.276 17.29 29.946 1.552-.896L18.467 1.38l-1.552.896ZM4.681 11.152H2.89v19.973h1.792V11.152Z"
                  fill="currentColor"
                />
                <path
                  d="M34.737 11.152h-1.792v19.973h1.792V11.152Z"
                  fill="currentColor"
                />
                <path
                  d="m18.41 37.813.784 1.355 15.043-8.685-.783-1.356-15.043 8.686Z"
                  fill="currentColor"
                />
                <path
                  d="M37.113 31.708a3.774 3.774 0 0 1-5.15 1.382 3.774 3.774 0 0 1-1.382-5.15 3.774 3.774 0 0 1 5.15-1.382 3.757 3.757 0 0 1 1.382 5.15ZM7.035 14.337a3.774 3.774 0 0 1-5.15 1.382 3.774 3.774 0 0 1-1.382-5.15 3.774 3.774 0 0 1 5.15-1.382 3.776 3.776 0 0 1 1.382 5.15ZM.514 31.708a3.776 3.776 0 0 1 1.382-5.15 3.776 3.776 0 0 1 5.15 1.382 3.776 3.776 0 0 1-1.382 5.15 3.771 3.771 0 0 1-5.15-1.382ZM30.592 14.337a3.776 3.776 0 0 1 1.382-5.15 3.776 3.776 0 0 1 5.15 1.382 3.776 3.776 0 0 1-1.382 5.15 3.774 3.774 0 0 1-5.15-1.382ZM18.813 42.278a3.764 3.764 0 0 1-3.768-3.768 3.764 3.764 0 0 1 3.768-3.768 3.764 3.764 0 0 1 3.768 3.768 3.771 3.771 0 0 1-3.768 3.768ZM18.813 7.536a3.764 3.764 0 0 1-3.768-3.768A3.764 3.764 0 0 1 18.813 0a3.764 3.764 0 0 1 3.768 3.768 3.764 3.764 0 0 1-3.768 3.768Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </a>
          <div
            class="MuiBox-root css-k62dlm"
          />
          <div
            class="MuiBox-root css-6su6fj"
          >
            <div
              class="MuiStack-root css-niqf4j-MuiStack-root"
            >
              <div
                class="MuiBox-root css-79elbk"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LanguageIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                href="/sign-in"
                tabindex="0"
              >
                sign in
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                href="/sign-up"
                tabindex="0"
              >
                sign up
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </header>
      <main
        class="MuiContainer-root MuiContainer-maxWidthMd css-x55gft-MuiContainer-root"
      >
        <div
          class="MuiStack-root css-120w1kz-MuiStack-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1k61bmk-MuiTypography-root"
          >
            Something went wrong
          </h2>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-7pb3oa-MuiSvgIcon-root"
            data-testid="WarningRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.47 21h15.06c1.54 0 2.5-1.67 1.73-3L13.73 4.99c-.77-1.33-2.69-1.33-3.46 0L2.74 18c-.77 1.33.19 3 1.73 3zM12 14c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
            />
          </svg>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="RefreshIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
            </span>
            Reload page
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </main>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          class="MuiStack-root css-1eoeowx-MuiStack-root"
        >
          <div
            class="MuiStack-root css-kdc3n8-MuiStack-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://github.com/Alyona8891"
            >
              <span
                class="MuiBadge-root css-1arcc7-MuiBadge-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-rounded css-12bapnq-MuiAvatar-root"
                >
                  <img
                    alt="Alyona's avatar"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://avatars.githubusercontent.com/u/79936182?v=4"
                  />
                </div>
                <span
                  class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-s1rlol-MuiBadge-badge"
                />
              </span>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://github.com/Lev1ossa"
            >
              <span
                class="MuiBadge-root css-1arcc7-MuiBadge-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-rounded css-12bapnq-MuiAvatar-root"
                >
                  <img
                    alt="Alexandr's avatar"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://avatars.githubusercontent.com/u/104268412?v=4"
                  />
                </div>
                <span
                  class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-s1rlol-MuiBadge-badge"
                />
              </span>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="https://github.com/lazy-goose"
            >
              <span
                class="MuiBadge-root css-3jxqn3-MuiBadge-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-rounded css-12bapnq-MuiAvatar-root"
                >
                  <img
                    alt="Maksim's avatar"
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://avatars.githubusercontent.com/u/119844669?v=4"
                  />
                </div>
                <span
                  class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-s1rlol-MuiBadge-badge"
                />
              </span>
            </a>
          </div>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-uwx3rb-MuiTypography-root-MuiLink-root"
            href="https://rs.school/react"
          >
            <time
              class="MuiBox-root css-1h14jth"
            >
              2024
            </time>
            <div
              class="MuiBox-root css-psrz38"
            >
              <span>
                RS
              </span>
              <span>
                SC HO OL
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
